src/ingest_validation_tools/schema_loader.py

```diff
@@ -22,139 +25,204 @@ class PreflightError(Exception):
     pass
 
 
-SchemaVersion = namedtuple('SchemaVersion', ['schema_name', 'version'])
+# TODO: why not a dataclass?
+SchemaVersion = namedtuple(
+    "SchemaVersion", ["schema_name", "version", "is_assay"], defaults=(True,)
+)
 
 
 def get_fields_wo_headers(schema: dict) -> List[dict]:
-    return [field for field in schema['fields'] if not isinstance(field, str)]
+    return [field for field in schema["fields"] if not isinstance(field, str)]
 
 
 def get_field_enum(field_name: str, schema: dict) -> List[str]:
     fields_wo_headers = get_fields_wo_headers(schema)
-    fields = [
-        field for field in fields_wo_headers
-        if field['name'] == field_name
-    ]
+    fields = [field for field in fields_wo_headers if field["name"] == field_name]
     if not fields:
         return []
     assert len(fields) == 1
-    return fields[0]['constraints']['enum']
+    return fields[0]["constraints"]["enum"]
 
 
-def get_table_schema_version_from_row(path: str, row: Dict[str, Any]) -> SchemaVersion:
-    '''
+def get_table_schema_version_from_row(
+    path: str, row: Dict[str, Any], cedar_validation: bool = True
+) -> SchemaVersion:
+    """
     >>> try: get_table_schema_version_from_row('empty', {'bad-column': 'bad-value'})
     ... except Exception as e: print(e)
-    empty does not contain "assay_type". Column headers: bad-column
-
-    >>> get_table_schema_version_from_row('v0', {'assay_type': 'PAS microscopy'})
-    SchemaVersion(schema_name='stained', version=0)
-
-    >>> try: get_table_schema_version_from_row('v42',
-    ... {'assay_type': 'PAS microscopy', 'version': 42})
+    empty does not contain "assay_type" or "dataset_type". Column headers: bad-column
+
+    >>> get_table_schema_version_from_row( \
+            'v0', \
+            {'assay_type': 'PAS microscopy'}, \
+            cedar_validation=False \
+        )
+    SchemaVersion(schema_name='stained', version='0', is_assay=True)
+
+    >>> try: get_table_schema_version_from_row('v42', \
+                {'assay_type': 'PAS microscopy', 'version': 42}, \
+                cedar_validation=False \
+                )
     ... except PreflightError as e: print(e)
     No schema where 'PAS microscopy' is assay_type and 42 is version
 
-    '''
-    if 'assay_type' not in row:
-        message = f'{path} does not contain "assay_type". '
-        if 'channel_id' in row:
-            message += 'Has "channel_id": Antibodies TSV found where metadata TSV expected.'
-        elif 'orcid_id' in row:
-            message += 'Has "orcid_id": Contributors TSV found where metadata TSV expected.'
+    """
+    assay = row["assay_type"] if "assay_type" in row else row.get("dataset_type")
```

Comment:
If the row contains neither one of these, then it will throw an exception. Might want to consider modifying so that there is a default False or "" value returned with row.get()